src/components/PostDetails.tsx

```diff
@@ -1,117 +1,128 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { ErrorNotification } from '../types/ErrorNotification';
+import { apiActions } from '../utils/apiAction';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post;
+  errorNotification: ErrorNotification;
+  setErrorNotification: (error: (prevError: ErrorNotification)
+  => ErrorNotification) => void;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost,
+  setErrorNotification,
+  errorNotification,
+}) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isAddComment, setIsAddComment] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    setIsAddComment(false);
+
+    apiActions.getPostComments(selectedPost.id)
+      .then(setComments)
+      .catch(() => setErrorNotification(
+        (prevErr: ErrorNotification) => ({ ...prevErr, comments: true }),
+      ))
+      .finally(() => setIsLoading(false));
+  }, [selectedPost]);
```

Comment:
![2023-10-18_15h50_21](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/61904995/8ba56b9f-f881-4fd9-ae97-b6f4e66523a1)